litex/soc/integration/soc_core.py

```diff
@@ -402,67 +402,69 @@ def check_regions_overlap(regions):
         return None
 
     def alloc_mem_region(self, name, length, type):
-        # Use type to limit search regions
-        search_regions = []
-        if "io" in type:
-            for _origin, _length in self.soc_io_regions.items():
-                search_regions.append(SoCMemRegion(origin=_origin, length=_length, type=type))
+        # Linker only regions have to be explicitly specified.
+        assert SoCMemRegion.Properties.linker_only not in type
+
+        CACHED_BOUNDRY = 0x80000000
```

Comment:
I think we should probably defined a `self.soc.cached_region` and a `self.soc.io_region` type things?

Reply:
We already have a `self.io_regions` and `self.soc_io_regions`: https://github.com/enjoy-digital/litex/blob/master/litex/soc/integration/soc_core.py#L113 but no `cached_region` since non `io_regions ` we considered cached. But we could add it if you think it's useful.